### ISSUE TITLE
Diff testing: after #39 run more test cases and iteratively improve

### ISSUE THREAD

**Issue by danwt** [cosmos/interchain-security#132]
Upon completion of

- https://github.com/cosmos/interchain-security/issues/39

the PR can be improved

- https://github.com/cosmos/interchain-security/pull/126

by running more test cases.

This will most likely reveal problem in the driver, so additional future work is expected.
